lib/internal/Magento/Framework/View/Asset/Repository.php

```diff
@@ -135,22 +135,17 @@ public function updateDesignParams(array &$params)
         }
 
         // Set themeModel
-        $theme = null;
         $area = $params['area'];
         if (!empty($params['themeId'])) {
-            $theme = $params['themeId'];
+            $params['themeModel'] = $this->getThemeById($params['themeId'], $area);
         } elseif (isset($params['theme'])) {
-            $theme = $params['theme'];
+            $params['themeModel'] = $this->getThemeByPath($params['theme'], $area);
         } elseif (empty($params['themeModel']) && $area !== $this->getDefaultParameter('area')) {
-            $theme = $this->design->getConfigurationDesignTheme($area);
+            $themeId = $this->design->getConfigurationDesignTheme($area);
+            $params['themeModel'] = $this->getTheme($themeId, $area);
         }
 
-        if ($theme) {
-            $params['themeModel'] = $this->getThemeProvider()->getThemeByFullPath($area . '/' . $theme);
-            if (!$params['themeModel']) {
-                throw new \UnexpectedValueException("Could not find theme '$theme' for area '$area'");
-            }
-        } elseif (empty($params['themeModel'])) {
+        if (empty($params['themeModel'])) {
             $params['themeModel'] = $this->getDefaultParameter('themeModel');
```

Comment:
For better readability please do `$themeModel = ...` instead of `$params['themeModel'] = ...` and only in the very end do `$params['themeModel'] = $themeModel ?: $this->getDefaultParameter('themeModel')`.